### DISCUSSION TITLE
Setting default printer per user

### DISCUSSION THREAD

**Discussion by mecrider** [ltsp/ltsp#520] [Q&A]
We are running LTSP on Ubuntu 20.04, using Gnome Shell desktops. We have multiple printers (all attached to the network) installed on a CUPS server, and all are visible to each client. Is there a way to set the default printer by client or user? Right now everyone's default printer is the system default printer on the CUPS server, and while some programs remember the last used printer, other programs go back to the default printer every time.

**Comment by alkisg** [comment]
To set a default printer you just need to set the PRINTER or the LPDEST environment variable: https://docs.oracle.com/cd/E36784_01/html/E36821/glldq.html

Currently, LTSP doesn't offer a way to set environment variables inside the user session; you'd need to use non-LTSP mechanisms for that. For example, if you put the following in ltsp.conf:

```shell
[mac:address]
POST_INIT_SET_PRINTER='echo export "PRINTER=HP-DesignJet-750c" > /etc/X11/Xsession.d/90ltsp'
```

...then this will set the default printer on a per-mac-address basis, IF the clients are using Xorg.

On the other hand, to set the default printer on a per-user basis, you could create a script named e.g. /usr/local/bin/mate-session, similar to this:

```shell
#!/bin/sh

case "$USER" in
john) export "PRINTER=HP-DesignJet-750c" ;;
bill) export "PRINTER=HP-DesignJet-1050c" ;;
esac

exec /usr/bin/mate-session "$@"
```

This will override your default session and set the PRINTER variable, then it will exec the real session.
In your case, the session is probably named gnome-session, not mate-session.

The same method can be used to set PRINTER on a mac-address basis on Wayland.

If we do find a better way to set session environment variables in a cross distro/environment manner, of course we can prefer that one...

**Reply by mecrider** [reply]
I ended up taking a route that was a little more tedious initially, but since it rarely has to be touched again it worked for me on Ubuntu 20.04/GNOME. I edited .profile in each user's home directory and added a line such as:
`export PRINTER=billing_laser`
